src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -201,6 +220,27 @@ def eq(self, y: Any):
 
         return True
 
+    def __getitem__(self, key: int) -> torch.Tensor:
+        """Allows to subset shares.
+
+        Args:
+            key (int): The share to be retrieved.
+
+        Returns:
+            share (torch.Tensor): Returned share.
+        """
+        return self.shares[key]
+
+    def __setitem__(self, key: int, newvalue: torch.Tensor) -> None:
+        """Allows to set share value to new value.
+
+        Args:
+            key (int): The share to be retrieved.
+            newvalue (torch.Tensor): New value of share.
+
+        """
+        self.shares = newvalue
```

Comment:
I think here it should be `self.shares[key] = newvalue`.

Reply:
You are right. Thanks.